python/xorbits/_mars/dataframe/window/rolling/aggregation.py

```diff
@@ -432,13 +440,19 @@ def tile(cls, op):
         if out.ndim == 1:
             params["shape"] = (inp.shape[0],)
         else:
-            params["shape"] = (inp.shape[0], params["shape"][1])
+            if op.func == "corr" and op.func_kwargs.get("pairwise", True):
```

Comment:
This line is same as line 407, we can check at beginning and record with a variable.

Reply:
Done.